code/game/gamemodes/game_mode.dm

```diff
@@ -326,91 +348,21 @@
-			for(var/job in restricted_jobs)					// Remove people who want to be antagonist but have a job already that precludes it
-				if(player.assigned_role == job)
-					candidates -= player
-					break
-
-	if(candidates.len < recommended_enemies)
-		for(var/mob/dead/new_player/player in players)
-			if(player.client && player.ready == PLAYER_READY_TO_PLAY)
-				if(!(role in player.client.prefs.be_special)) // We don't have enough people who want to be antagonist, make a separate list of people who don't want to be one
-					if(!jobban_isbanned(player, "Syndicate") && !jobban_isbanned(player, role)) //Nodrak/Carn: Antag Job-bans
-						drafted += player.mind
-
-	if(restricted_species)
-		for(var/mob/dead/new_player/player in drafted)
-			for(var/species in restricted_species)
-				if(player.client.prefs.pref_species.id == species)
-					drafted -= player
-					break
-
-	if(restricted_jobs)
-		for(var/datum/mind/player in drafted)				// Remove people who can't be an antagonist
-			for(var/job in restricted_jobs)
-				if(player.assigned_role == job)
-					drafted -= player
-					break
-
-	drafted = shuffle(drafted) // Will hopefully increase randomness, Donkie
-
-	while(candidates.len < recommended_enemies)				// Pick randomlly just the number of people we need and add them to our list of candidates
-		if(drafted.len > 0)
-			applicant = pick(drafted)
-			if(applicant)
-				candidates += applicant
-				drafted.Remove(applicant)
-
-		else												// Not enough scrubs, ABORT ABORT ABORT
-			break
-
-	if(restricted_species)
-		for(var/mob/dead/new_player/player in drafted)
-			for(var/species in restricted_species)
-				if(player.client.prefs.pref_species.id == species)
-					drafted -= player
-					break
-
-	if(restricted_jobs)
-		for(var/datum/mind/player in drafted)				// Remove people who can't be an antagonist
-			for(var/job in restricted_jobs)
-				if(player.assigned_role == job)
-					drafted -= player
-					break
-
-	drafted = shuffle(drafted) // Will hopefully increase randomness, Donkie
-
-	while(candidates.len < recommended_enemies)				// Pick randomlly just the number of people we need and add them to our list of candidates
-		if(drafted.len > 0)
-			applicant = pick(drafted)
-			if(applicant)
-				candidates += applicant
-				drafted.Remove(applicant)
-
-		else												// Not enough scrubs, ABORT ABORT ABORT
-			break
-
-	return candidates		// Returns: The number of people who had the antagonist role set to yes, regardless of recomended_enemies, if that number is greater than recommended_enemies
-							//			recommended_enemies if the number of people with that role set to yes is less than recomended_enemies,
-							//			Less if there are not enough valid players in the game entirely to make recommended_enemies.
+				candidates_preferred += player.mind
+			else
+				candidates_nothanks += player.mind
 
+	if(candidates_preferred.len < recommended_enemies)
+		for(var/datum/mind/player in candidates_nothanks)

```

Comment:
Uh won't this start adding job-banned and ineligible players if there are not enough recommended enemies?

Reply:
No, ineligble players won't be in either list. `candidate_nothanks` just holds people who do not have antag enabled in their preferences (force antag). That's checked here- https://github.com/OracleStation/OracleStation/pull/684/files#diff-2334d1bad599b5c51def0eeaa65d7bfdR355

Reply:
Ah right I was reading the indentation incorrectly, Thought that if that proc returned false then it would add them to the `candidate_nothanks` list.